### DISCUSSION TITLE
How to use old version or another chrome driver?

### DISCUSSION THREAD

**Discussion by ne-skazhu** [seleniumbase/SeleniumBase#1892] [Q&A] [answered]
I couldn't use the old version of chrome-driver until I restricted the rights to this file. Is a more conservative method implemented? I need support for Yandex Browser, which often uses an older version of chrome, and when running tests, seleniumbase tries to update the version to the latest.  And I can't use the original name yandex-driver.exe, because you have a lot of binding to the name in the code

**Comment by mdmintz** [comment, marked as answer]
You can use Syntax Format 9 (or 10) from https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#sb_sf_09, which gives you total control with how you spin up your web browsers. This is especially useful when you want to use a browser that isn't officially supported.

In some situations, you might want to change the Chromium browser binary location (when using the other syntax formats), and you can do that with: ``pytest --binary-location=PATH``.

**Reply by ne-skazhu** [reply]
Thank you! It would be great if you add parameter of automatic update driver to the configuration file in future versions. There are situations when the browser does not have time to update behind the driver and you will need to save the working version.
Thank you very much for your framework, found it just recently and it's a pleasure to use it!

**Reply by ne-skazhu** [reply]
Oh, I'm sorry to ask all in one place, but it would also be great if you make a built-in ability to override the default launch of the browser for each test, now to run a specific test using the mobile version, I use this code 
`self.get_new_driver(browser=self.browser, is_mobile=True)`
, which is not very beautiful , in addition, while the test is running, a second window is created, which adds time to the test.
From the documentation, I found only the launch of the mobile version only through the start of `pytest --mobile` , but this option is inconvenient for me.

By the way, is there any life hack for cross-browser testing? So far, I have defined for myself the means of launching pytest, but it would be nice to indicate this right inside the tests.

**Reply by mdmintz** [reply]
For cross-browser testing, you can set the browser you want from the command-line with ``pytest``, eg: ``--chrome``, ``--edge``, ``--firefox``, ``--safari``. (``--chrome`` is the default option)
